server/src/main/java/org/opensearch/index/mapper/NumberFieldMapper.java

```diff
@@ -841,16 +995,24 @@ public BigInteger parse(XContentParser parser, boolean coerce) throws IOExceptio
             }
 
             @Override
-            public Query termQuery(String field, Object value) {
+            public Query termQuery(String field, Object value, boolean hasDocValues, boolean isSearchable) {
                 if (hasDecimalPart(value)) {
                     return Queries.newMatchNoDocsQuery("Value [" + value + "] has a decimal part");
                 }
                 BigInteger v = parse(value, true);
+                if (isSearchable && hasDocValues) {
+                    Query query = BigIntegerPoint.newExactQuery(field, v);
+                    Query dvQuery = SortedUnsignedLongDocValuesRangeQuery.newSlowRangeQuery(field, v, v);
```

Comment:
I know that `newSlowExactQuery` would be delegating to range query, but the implementation would be very consistent with Long / Int / Float / ... 

```suggestion
                    Query dvQuery = SortedUnsignedLongDocValuesSetQuery.newSlowExactQuery(field, v);
```